### PR TITLE
POC: verify tox upgrading on zuul

### PR DESCRIPTION
**Pull request by ssbarnea**
Depends-On: https://github.com/ansible/project-config/pull/400

### CODE REVIEW

File: tools/test-setup.sh [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #!/bin/bash
-set -Eeuxo pipefail
+set -euxo pipefail
```
Comment (webknjaz) [inline tools/test-setup.sh]: I'd rather not remove `-E`

File: tools/test-setup.sh [hunk 2]
```diff
@@ -8,4 +8,8 @@ set -Eeuxo pipefail
 # non-sudo mode.
 PYTHON=$(command -v python3 python | head -n1)
 
-sudo "$PYTHON" -m pip install -U tox "zipp<0.6.0;python_version=='2.7'"
+# $PYTHON -m pip install --user "tox>=3.8.0" "zipp<0.6.0;python_version=='2.7'"
```
Comment (webknjaz) [inline tools/test-setup.sh]: Upgrade virtualenv before doing anything else. Use a separate `pip install` command.

File: tox.ini [hunk 3]
```diff
@@ -5,6 +5,7 @@ isolated_build = true
 requires =
   setuptools >= 41.4.0
   pip >= 19.3.0
+  six
```
Comment (webknjaz) [inline tox.ini]: I don't think it's needed.

File: tools/test-setup.sh [hunk 4]
```diff
@@ -8,4 +8,10 @@ set -Eeuxo pipefail
 # non-sudo mode.
 PYTHON=$(command -v python3 python | head -n1)
 
-sudo "$PYTHON" -m pip install -U tox "zipp<0.6.0;python_version=='2.7'"
+$PYTHON -m pip install --user "virtualenv>=20.0.10"
+$PYTHON -m pip install --user "tox>=3.8.0" "zipp<0.6.0;python_version=='2.7'"
```
Comment (webknjaz) [inline tools/test-setup.sh]: why is zipp needed?

Comment (ssbarnea) [reply tools/test-setup.sh]: There is a bug in older versions of pip where it installs a version incompatible with py27.

BTW, this PR is not ready for review yet, I am using it to test various things. I will remove draft status when I consider it ready.

**Comment by ssbarnea** [comment]
recheck

**Pull request closed without merging**